webhook_features/src/features/summary_comment.rs

```diff
@@ -0,0 +1,631 @@
+                                    .or_insert_with(Vec::new)
+                                    .push((ack.user.clone(), ack.url.clone())); // Store the user and the URL of the stale ACK
+                            }
+                            return acc;
+                        }
+                        if ack.ack_type.requires_commit_hash() && ack.commit.is_none() {
+                            // ACK requires a commit hash but none is referenced
+                            return acc;
+                        }
+
+                        acc.entry(ack.ack_type)
+                            .or_insert_with(Vec::new)
+                            .push((ack.user.clone(), ack.url.clone())); // Store the user and the URL of the ACK
+                        acc
+                    });
+
+                // Display ACKs in the following order
+                for ack_type in vec![
+                    AckType::ACK,
+                    AckType::NACK,
+                    AckType::ConceptACK,
+                    AckType::ConceptNACK,
+                    AckType::CodeReviewACK,
+                    AckType::CodeReviewNACK,
+                    AckType::ApproachACK,
+                    AckType::ApproachNACK,
+                    AckType::StaleACK,
+                ] {
+                    if let Some(users) = ack_map.get(&ack_type) {
+                        let mut users = users.clone();
+                        users.sort();
+                        comment += &format!(
+                            "| {} | {} | {} |\n",
+                            ack_type.to_string(),
+                            users.len(),
+                            users
+                                .iter()
+                                .map(|(user, url)| format!("[{}]({})", user, url))
+                                .collect::<Vec<String>>()
+                                .join(", ")
+                        );
+                    }
+                }
+            }
+            None => {}
+        }
+
+        comment += "\n";
+    }
+
+    // TODO: PR Conflicts
+
+    comment
+}
+
+async fn create_summary_comment(
+    payload: &serde_json::Value,
+    octocrab: &octocrab::Octocrab,
+) -> Result<()> {
+    let owner = {
+        let login = &payload["pull_request"]["user"]["login"];
+        login
+            .as_str()
+            .ok_or(DrahtBotError::InvalidLogin(login.to_string()))?
+    };
+
+    let repo_name = {
+        let name = &payload["repository"]["name"];
+        name.as_str()
+            .ok_or(DrahtBotError::InvalidRepositoryName(name.to_string()))?
+    };
+
+    let pr_number = {
+        let pr_number = &payload["number"];
+        pr_number
+            .as_u64()
+            .ok_or(DrahtBotError::InvalidPullRequestNumber(
+                pr_number.to_string(),
+            ))?
+    };
+
+    let cmt = util::get_metadata_sections(&octocrab, &octocrab.issues(owner.clone(), repo_name.clone()), pr_number).await?;
```

Comment:
any reason to clone this here, when a reference can be passed instead?

Reply:
got this
```
   --> src/features/summary_comment.rs:217:71
    |
217 |     let cmt = util::get_metadata_sections(&octocrab, &octocrab.issues(&owner, &repo_name), pr_number).await?;
    |                                                                ------ ^^^^^^
    |                                                                |      |
    |                                                                |      the trait `std::convert::From<&&str>` is not implemented for `std::string::String`
    |                                                                |      help: consider dereferencing here: `&*owner`
    |                                                                required by a bound introduced by this call
    |
```

Reply:
The type of owner is already `&str`, see https://docs.rs/serde_json/latest/serde_json/value/enum.Value.html#method.as_str

So adding `&` will make it `&&str`, which wouldn't work

Reply:
Right, not sure why I did that, thanks.